### PR TITLE
Improved iLQS

### PR DESCRIPTION
**Pull request by thowell**
- Improved logic and compute inverse mapping for iLQS planner
- Actuator force cost for hand task

### CODE REVIEW

File: mjpc/tasks/hand/hand.cc [hunk 1]
```diff
@@ -27,11 +27,10 @@ std::string Hand::XmlPath() const {
 std::string Hand::Name() const { return "Hand"; }
 
 // ---------- Residuals for in-hand manipulation task ---------
-//   Number of residuals: 5
+//   Number of residuals: 4
 //     Residual (0): cube_position - palm_position
 //     Residual (1): cube_orientation - cube_goal_orientation
 //     Residual (2): cube linear velocity
-//     Residual (3): cube angular velocity
 //     Residual (4): control
```
Comment (yuvaltassa) [inline mjpc/tasks/hand/hand.cc:34]: 3

File: mjpc/tasks/hand/task.xml [hunk 2]
```diff
@@ -39,6 +39,7 @@
     <user name="In Hand" dim="3" user="1 20 0 100 0.02 2"/>
     <user name="Orientation" dim="3" user="0 3 0 10" />
     <user name="Cube Vel." dim="3" user="0 10 0 20" />
+    <user name="Actuator" dim="20" user="0 1.0e-1 0.0 1.0" />
```
Comment (yuvaltassa) [inline mjpc/tasks/hand/task.xml:42]: 1.0e-1 -> 0.1

File: mjpc/planners/ilqs/planner.cc [hunk 3]
```diff
@@ -135,21 +173,36 @@ void iLQSPlanner::OptimizePolicy(int horizon, ThreadPool& pool) {
...
+    }
+    ilqg.model_derivative_compute_time = 0.0;
+    ilqg.cost_derivative_compute_time = 0.0;
+    ilqg.backward_pass_compute_time = 0.0;
+    ilqg.rollouts_compute_time = 0.0;
+    ilqg.policy_update_compute_time = 0.0;
+
+    // set new active policy 
+    active_policy = kSampling;
+    return;
```
Comment (yuvaltassa) [inline mjpc/planners/ilqs/planner.cc:192]: // best rollout is from sampling, terminate early

**Review by yuvaltassa — CHANGES_REQUESTED** [review]

File: mjpc/tasks/hand/hand.cc (hunk 1 above)
Comment (thowell) [reply mjpc/tasks/hand/hand.cc]: Done.

File: mjpc/tasks/hand/task.xml (hunk 2 above)
Comment (thowell) [reply mjpc/tasks/hand/task.xml]: Done.

File: mjpc/planners/ilqs/planner.cc (hunk 3 above)
Comment (thowell) [reply mjpc/planners/ilqs/planner.cc:194]: Done.

**Comment by thowell** [comment]
Comments addressed.

**Pull request merged**